File: Core/Models/CRUDExample.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Core.Models
{
    [Table("CRUDExample")]
    public class CRUDExample
    {
        [Key]
        public int Id { get; set; }
        public string Text { get; set; }
        public DateTimeOffset DateCreated { get; set; }
        public bool IsDeleted { get; set; }
        public DateTimeOffset? DateDeleted { get; set; }
    }
}

File: UI.Web/ClientApp/src/app/models/CRUDExample.ts
export class CRUDExample {
    constructor(){
        this.id = 0;
        this.text= null;
        this.dateCreated=  null;
        this.dateDeleted=  null;
        this.isDeleted=  false;
    }

    id: number;
    text: string;
    dateCreated: Date;
    isDeleted: boolean;
    dateDeleted: Date;
}